### ISSUE TITLE
False positive when checking for duplicate properties

### ISSUE THREAD

**Issue by 0Neji** [stylelint/stylelint#4763]
> Clearly describe the bug

Running through setup of a new project and tweaking the configuration to my preferences. All is looking good until I get to the `declaration-block-no-duplicate-properties` - I have a very simple test stylesheet which throws and error when processing that I have a duplicate `display` properties which, as far as I can tell, isn't the case.

> Which rule, if any, is the bug related to?

e.g. `declaration-block-no-duplicate-properties`

> What code is needed to reproduce the bug?

```css
:root {
    --linkColour: purple;
}

a {
    color: var(--linkColour);
    margin: 1px 1px 1px 1px;
}

.item {
    display: flex;
}

::placeholder {
    color: gray;
    font-family: 'Times', 'Times', serif;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
    "rules": {
        "color-no-invalid-hex": true,
        "font-family-no-duplicate-names": true,
        "font-family-no-missing-generic-family-keyword": true,
        "function-calc-no-invalid": true,
        "function-calc-no-unspaced-operator": true,
        "function-linear-gradient-no-nonstandard-direction": true,
        "string-no-newline": true,
        "unit-no-unknown": true,
        "property-no-unknown": true,
        "keyframe-declaration-no-important": true,
        "declaration-block-no-duplicate-properties": true,
        "declaration-block-no-shorthand-property-overrides": true,
        "block-no-empty": true,
        "selector-pseudo-class-no-unknown": true,
        "selector-pseudo-element-no-unknown": true,
        "media-feature-name-no-unknown": true,
        "at-rule-no-unknown": true,
        "comment-no-empty": true,
        "no-descending-specificity": true,
        "no-duplicate-at-import-rules": true,
        "no-duplicate-selectors": true,
        "no-empty-source": true,
        "no-extra-semicolons": true,
        "no-invalid-double-slash-comments": true,
        "shorthand-property-no-redundant-values": true
    }
}
```

> Which version of stylelint are you using?

`13.3.3`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Running through gulp:

```
var postcss = require('gulp-postcss');

var processors = [
    require('postcss-cssnext')(),
    require('cssnano')(),
    require("stylelint")()
];

... paths cut for brevity

gulp.task('styles', function () {
    return gulp.src('./src/css/main.css')
        .pipe(postcss(processors))
        .pipe(gulp.dest(basePaths.dist + 'assets/css/'))
        .pipe(gulp.dest(basePaths.cms + 'assets/css/'))
});
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Nope, pure CSS.

> What did you expect to happen?

No warnings/errors to be flagged

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
[14:39:36] gulp-postcss: main.css
stylelint: D:\Repos\Freelance\Umble\umble\frontend\src\css\main.css:11:5: Unexpected duplicate "display" (declaration-block-no-duplicate-properties)
stylelint: D:\Repos\Freelance\Umble\umble\frontend\src\css\main.css:11:5: Unexpected duplicate "display" (declaration-block-no-duplicate-properties)
```

**Comment by m-allanson** [comment]
Hey @0Neji, I'm [unable to replicate this on the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcCBOEIBcAEwA6A7DBGAtEQDYCWuA1gMISkQCuS8GADs26TANx4C+ePAEMseQhkgMWGAG7CkAChIVqdBswCUffIQC2CgOaVWARjYAPDOas3rlnYNx4AdOTQw9Y3QSjkAzlzCAJ6sAGbcFo5CuPDwQWAwABb0sEjeElIQMoZIIToSYRC4aERhwnrkpKEYAOQAKuR6MP61ADR1jc2tHf4wSORhjiBtIP7BJcIWCOD+-iPgxWHkhjM4PhjYIEiM3P5brOsSEltZSES4EESU8hRQREkwFgcYaDswbeLHBFtFJWUVKrBC5XKCMLjkMDCDwXCotF5vRgfL7fX7FUrlSrVEFESpzSiGIiGGC4fqQgFY4FUGDBADu2SgCPenw2JxAYUYuDAaHIxSIUNIYBxN2EdyZSJZ3x+7M53N5uH5oqFlyInMCwkS9wgbH60Oy4uRrMIv1lPL5qhgCiJeT8JNKKsuuH8aGEuCgCnufiQMDlxQNkqlW2dA1whIdMFpFv9KOOW057hxnKol1pzgQr2ZMbZbBQOqQaGBKqTKbTrERhqlmxA1OCYTyzSIsDApAU0Plwr0bGyLpK0aN0qbLbyZoVACMGGAqDiwRCoTCc9r+jz4enywHUSBB62R0RxxBJzj-Cl80lXfcF3mC0QILJ+gNYPtV5n+1W9weVZ42AW+5WgzBuNy2REGwfSMFAVzNsIcyJtQJY-oGYz-j6aBASBMBgVcSHNP8RawRAqbwRuzR+MIZSWmgzAwLCDa4cm+GlhmEpZsaIDQkQOzcDBdEEU+TEvqcEB6Nh9qYZ2368RWCEqg+iRugSRCBD6gyQu4wSEbGIDSeCFBzlRbFNF2+bsbsK5ls+v6aaC2mQtCVF9ABKFII+Zl8RZH5icC-hMEgiTqWyH4WG8pF9JUWROn5LEqiKdyNkw452S2R78oJwnOYxkkbke3anm6wG5kuhZXN6YJuq6pS3EiaXljGTj8CA-BAA).

I suspect the issue you're seeing is because you have stylelint running _after_ cssnext and cssnano.

Can you try changing your processor config like:

```diff
var processors = [
+    require("stylelint")(),
    require('postcss-cssnext')(),
    require('cssnano')(),
-    require("stylelint")()
];
```

Check out the docs for [more info on running stylelint as a PostCSS plugin](https://stylelint.io/user-guide/usage/postcss-plugin).

**Comment by 0Neji** [comment]
Thanks @m-allanson - you're absolutely right! Hadn't thought about the order being important but that's done the trick.

Thanks for the super speedy reply - and apologies for the n00b question!

**Comment by m-allanson** [comment]
No problem, I'm glad I could help :)
